src/FLACTagReader.js

```diff
@@ -0,0 +1,344 @@
+   * More info on the FLAC specification may be found here:
+   * https://xiph.org/flac/format.html
+   * @param {MediaFileReader} mediaFileReader - The MediaFileReader used to parse the file.
+   * @param {number} offset - The offset to start checking the header from.
+   * @param {LoadCallbackType} callbacks - The callback to call once the header has been found.
+   */
+  _loadBlock(
+    mediaFileReader: MediaFileReader,
+    offset: number,
+    callbacks: LoadCallbackType
+  ) {
+    var self = this;
+    /* As mentioned above, this first byte is loaded to see what metadata type
+     * this block represents.
+     */
+    var blockHeader = mediaFileReader.getByteAt(offset);
+    /* The last three bytes (integer 24) contain a value representing the length
+     * of the following metadata block. The 1 is added in order to shift the offset
+     * by one to get the last three bytes in the block header.
+     */
+    var blockSize = mediaFileReader.getInteger24At(offset + 1, true);
+    /* This conditional checks if blockHeader (the byte retrieved representing the
+     * type of the header) is one the headers we are looking for.
+     *
+     * If that is not true, the block is skipped over and the next range is loaded:
+     * - offset + 4 + blockSize adds 4 to skip over the initial metadata header and
+     * blockSize to skip over the block overall, placing it at the head of the next
+     * metadata header.
+     * - offset + 4 + 4 + blockSize does the same thing as the previous block with
+     * the exception of adding another 4 bytes to move it to the end of the new metadata
+     * header.
+     */
+    if (COMMENT_HEADERS.indexOf(blockHeader) !== -1) {
+      /* 4 is added to offset to move it to the head of the actual metadata.
+       * The range starting from offsetMatadata (the beginning of the block)
+       * and offsetMetadata + blockSize (the end of the block) is loaded.
+       */
+      var offsetMetadata = offset + 4;
+      mediaFileReader.loadRange([offsetMetadata, offsetMetadata + blockSize], {
+        onSuccess: function() {
+          self._commentOffset = offsetMetadata;
+          self._nextBlock(mediaFileReader, offset, blockHeader, blockSize, callbacks);
+          // mediaFileReader.loadRange([offset + 4 + blockSize, offset + 4 + 4 + blockSize], {
+          //   onSuccess: function() {
+          //     self._loadBlock(mediaFileReader, offset + 4 + blockSize, callbacks);
+          //   }
+          // });
+          // callbacks.onSuccess();
+        }
+      });
+    } else if (PICTURE_HEADERS.indexOf(blockHeader) !== -1) {
+      var offsetMetadata = offset + 4;
+      mediaFileReader.loadRange([offsetMetadata, offsetMetadata + blockSize], {
+        onSuccess: function() {
+          self._pictureOffset = offsetMetadata;
+          self._nextBlock(mediaFileReader, offset, blockHeader, blockSize, callbacks);
+          // mediaFileReader.loadRange([offset + 4 + blockSize, offset + 4 + 4 + blockSize], {
```

Comment:
Not really, that was the original code for loading a new range until the Vorbis comment block was identified. I kept it there so I had it for reference until _nextBlock would work. I'll remove it on the next commit.